cooked-validators/src/Cooked/Attack.hs

```diff
@@ -0,0 +1,158 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cooked.Attack where
+
+import Cooked.MockChain
+import Cooked.Tx.Constraints
+import Cooked.Tx.Constraints.Optics
+import Data.Maybe
+import qualified Ledger as L hiding (validatorHash)
+import qualified Ledger.Typed.Scripts as L
+import Optics.Core
+import qualified PlutusTx as Pl
+import qualified PlutusTx.Numeric as Pl (negate)
+import Type.Reflection
+
+-- * The type of attacks, and functions to turn optics into attacks
+
+-- | The type of attacks that operate on a single transaction. The idea is to
+-- try to modify a transaction, or return @Nothing@ if the modification does not
+-- apply; use this with the 'somewhere' and 'everywhere' modalities from
+-- 'Cooked.MockChain.Monad'.
+type Attack = TxSkel -> Maybe TxSkel
+
+-- | The simplest way to make an attack from an optic: Try to apply the given
+-- function to all of the optic's foci. Fail, returning @Nothing@, if nothing is
+-- in focus or if all foci are evaluated to @Nothing@.
```

Comment:
I don't understand your question about returning more than one transaction. The `dupTokenAttack` returns a `Maybe TxSkel`, so at most one transaction. There is, in fact an example for the behaviour you describe in the [unit test for `dupTokenAttack`](https://github.com/tweag/plutus-libs/blob/aa7ebb198bf65fc9f583425b614d32e43249f1b2/cooked-validators/tests/Cooked/AttackSpec.hs#L102).

I agree that a comment to the effect that `mkAttack` returns `Just` iff at least one of the foci returns `Just` is due.

Reply:
> I agree that a comment to the effect that `mkAttack` returns `Just` iff at least one of the foci returns Just is due

And `mkAttack` performs the modifications on all foci that the auxiliary function returned `Just`. 

My "question" was ill-phrased: my concern is that I think this needs to be commented out, especially because when writing something like `somewhere (dupTokenAttack f)`, where `f` has multiple foci, it could get confusing pretty fast.